### PR TITLE
Feature/filter function by category

### PR DESCRIPTION
**Pull request by Jessica-Windoffer**

### CODE REVIEW

File: components/FilterByCategory.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+            style={{
+              color: selectedByIcon.includes("Others") ? "orange" : "black",
+            }}
+          >
+            <BackgroundIcons>
+              <IconOthers />
+            </BackgroundIcons>
+            <LabelIcons>Others</LabelIcons>
+          </ContainerIconLabel>
+        </Row>
```
Comment (F-Kirchhoff) [inline components/FilterByCategory.js:183]: This is a lot of repetitive code. Can you find a solution where you map over a single array of categories? Hint: you would need to get rid of your "Row" component and display the elements in a grid or wrapping flexbox.

File: components/FormCreate.js [hunk 2]
```diff
@@ -199,13 +203,35 @@ export default function FormCreate({ onAddActivity }) {
...
+            <option value="Extrem|Paragliding">Paragliding</option>
+            <option value="Extrem|Bungeejumping">Bungeejumping</option>
+            <option value="Extrem|Motorcross">Motorcross</option>
+            <option value="Extrem|Rafting">Rafting</option>
+
+            <option value="Indoor|Curling">Curling</option>
+            <option value="Indoor|Bouldern">Bouldern</option>
+            <option value="Indoor|Ice Hockey">Ice Hockey</option>
+
+            <option value="Others|Others">Others</option>
```
Comment (F-Kirchhoff) [inline components/FormCreate.js:234]: is there a specific reason why you use a "|"? If not please use a "_" or "-". Special characters can lead to weird bugs if you are not cautious.

File: pages/activityList.js [hunk 3]
```diff
@@ -79,22 +80,36 @@ const ErrorMessage = styled.p`
 
 export default function ActivityList({ activities }) {
   const [searchTerm, setSearchTerm] = useState("");
+  const [selectedByIcon, setSelectedByIcon] = useState([]);
+  console.log(selectedByIcon);
```
Comment (F-Kirchhoff) [inline pages/activityList.js:84]: please remove all logs

File: pages/activityList.js [hunk 4]
```diff
@@ -79,22 +80,36 @@ const ErrorMessage = styled.p`
...
+      filtered = filtered.filter((activity) =>
+        selectedByIcon.includes(activity.categoryFilter)
+      );
+    }
+
+    return filtered;
   };
 
-  const filteredActivities = getFilteredActivities(searchTerm, activities);
+  const filteredActivities = getFilteredActivities();
```
Comment (F-Kirchhoff) [inline pages/activityList.js:104]: if you don't plan to use the function somewhere else I would suggest to remove the function and put the code directly into your component. This is a bit less convoluted.

File: pages/activityList.js [hunk 5]
```diff
@@ -79,22 +80,36 @@ const ErrorMessage = styled.p`
...
+
+  const handleIconClick = (category) => {
+    if (selectedByIcon.includes(category)) {
+      setSelectedByIcon(selectedByIcon.filter((c) => c !== category));
```
Comment (F-Kirchhoff) [inline pages/activityList.js:108]: please don't use abbreviations like "c"

**Review by F-Kirchhoff — CHANGES_REQUESTED** [review]
Hey Activity Team,
very nice PR already, I like the grid of filter buttons that you came up with. Please take a look at the comments.

File: components/FilterByCategory.js [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+  flex-direction: column;
+  align-items: center;
+  text-decoration: none;
+
+  &:active {
+    color: black;
+  }
+`;
+
+export default function FilterByCategory({ handleIconClick, selectedByIcon }) {
```
[the remark this answers is not in the document]
Comment (Jessica-Windoffer) [reply components/FilterByCategory.js:81]: ✅

File: components/FormCreate.js (hunk 2 above)
Comment (Jessica-Windoffer) [reply components/FormCreate.js:234]: ✅

File: pages/activityList.js (hunk 5 above)
Comment (Jessica-Windoffer) [reply pages/activityList.js:108]: ✅

File: pages/activityList.js (hunk 4 above)
Comment (Jessica-Windoffer) [reply pages/activityList.js:104]: This is a point that is not implementable as we had a problem yesterday where the activities list was shown twice up to the point when we implemented the `const getFilteredActivities()`in this way. That is why this function needs to stay here. 😊

File: components/FilterByCategory.js (hunk 1 above)
Comment (Jessica-Windoffer) [reply components/FilterByCategory.js:183]: ✅

File: components/FilterByCategory.js [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+  return (
+    <>
+      <IconsContainer>
+        <Row>
+          <ContainerIconLabel
+            onClick={() => handleIconClick("Water")}
+            style={{
+              color: selectedByIcon.includes("Water") ? "orange" : "black",
+            }}
```
[the remark this answers is not in the document]
Comment (Jessica-Windoffer) [reply components/FilterByCategory.js:90]: ✅

File: components/CatetoryFilters.js [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+const categories = [
+  { name: "Water", icon: <IconWater /> },
+  { name: "Mountains", icon: <IconMountains /> },
+  { name: "Winter", icon: <IconWinter /> },
+  { name: "Biking", icon: <IconBiking /> },
+  { name: "Hiking", icon: <IconHiking /> },
+  { name: "Extrem", icon: <IconExtreme /> },
+  { name: "Indoor", icon: <IconIndoor /> },
+  { name: "Others", icon: <IconOthers /> },
+];
```
Comment (F-Kirchhoff) [inline components/CatetoryFilters.js:60]: nice ✨

File: components/CatetoryFilters.js [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+        <ContainerIconLabel
+          key={category.name}
+          onClick={() => handleIconClick(category.name)}
+          selected={selectedByIcon.includes(category.name)}
+        >
+          <BackgroundIcons>{category.icon}</BackgroundIcons>
+          <LabelIcons>{category.name}</LabelIcons>
+        </ContainerIconLabel>
+      ))}
+    </IconsContainer>
```
Comment (F-Kirchhoff) [inline components/CatetoryFilters.js:75]: this so so much cleaner. Great job!

**Review by F-Kirchhoff — APPROVED** [review]
LGTM

**Pull request merged**